### PR TITLE
Add an option to use other compilers, with specific options for Jiphy

### PR DESCRIPTION
**Pull request by DuncanMacWeb**
Users can now specify Jiphy instead of Transcrypt if they want in webpack.config.js

### CODE REVIEW

**Comment by martim00** [comment]
Great contribution! I'm going to publish to npm asap.

File: index.js [hunk 1]
```diff
@@ -1,30 +1,65 @@
...
+    const compilers = {
+        transcrypt: {
+            switches: '-b -n',
+            folder: `.${slash}__javascript__`,
+            install: 'pip install transcrypt',
+            python_version: '3.x'
+        },
+        jiphy: {
+            switches: '',
+            folder: `.${slash}`,
```
Comment (DuncanMacWeb) [inline index.js:23]: @martim00 Just realised this should be `folder: '.'` as the `${slash}` is appended on line 55 below. Happy to submit another PR?

File: index.js [hunk 2]
```diff
@@ -1,30 +1,65 @@
...
         callback(null, source);
     }
 
-    cmd.get('transcrypt -b -n ' + entry, function(err, data, stderr) {
+    cmd.get(`${compiler.name} ${compiler.switches} ${name}.py`, function(err, data, stderr) {
         if (!err) {
-            js = fs.readFileSync("./__javascript__/" + name + ".js", "utf8")
+            const filename = `${compiler.folder}${slash}${name}.js`;
+            js = fs.readFileSync(filename, "utf8");
+            fs.unlinkSync(filename);
```
Comment (DuncanMacWeb) [inline index.js:57]: @martim00 so on line 57 the output file the compiler generates gets deleted. To reduce the risk of data loss should we make sure the compiler’s output goes into a temporary folder instead?

**Review by DuncanMacWeb — COMMENTED** [review]
Just a couple of things that have come to my attention — might be worth addressing before you publish to npm?

File: index.js (hunk 1 above)
Comment (martim00) [reply index.js:23]: Sure, you're welcome!

File: index.js (hunk 2 above)
Comment (martim00) [reply index.js:57]: Yeah, I think it could be better. But since it depends on the compiler switch to change the output folder we'll have to expose it in the compiler settings. Do you think it pays the tradeoff to have another option in the settings? Or maybe we can live with that since it is already a tmp generated file? 

Also, sorry for the delay in response. I think I'll have a time to work in this project by the weekend. I'm experimenting with javascript dependencies inside python code (https://github.com/martim00/python-webpack-loader/issues/2). I don't know if jiphy has this problem. What I need is way to call a javascript library inside python code. Like using axios, for example. 

```python
class Foo:
   def call_api(self):
       # using js_get is the way to tell transcrypt to keep 'get' in the generated javascript
       axios.js_get("www.google.com")
```

If I just use the axios api the webpack does not include it in the python module dependencies and we get a undefined axios. I think it is because we have to specify the es6 import 'axios' at the top of the python file. Or we could use a solution like https://github.com/webpack-contrib/css-loader which transform any css @import syntax into js import/require so the webpack can understand that the css file depends on another and bundle it together.

Do you have any suggestions on that?
